util/metrics/prometheus.py

```
@@ -162,23 +222,93 @@ def run(self):
+    # Try to parse from request path
+    try:
+        path = request.path.strip("/")
+        if "/" in path:
+            # Try to parse as namespace/repository
+            parts = path.split("/", 1)
+            if len(parts) >= 1:
+                potential_namespace = parts[0]
+                # Basic validation: namespace should be alphanumeric with dashes/underscores
+                if potential_namespace and all(c.isalnum() or c in "-_" for c in potential_namespace):
+                    return potential_namespace
+    except:
```

Ruff (0.14.6)

Do not use bare `except`

(E722)